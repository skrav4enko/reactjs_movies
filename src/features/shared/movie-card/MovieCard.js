import React from 'react';
import './MovieCard.scss';

function MovieCard(props) {
  const { movie, navigateTo } = props;

  return (
    <div className="movie-card" onClick={() => navigateTo(movie.id)}>
      <div className="movie-card__poster">
        <img className="movie-card__img" src={movie.poster_path} alt={movie.title} />
      </div>
      <div className="movie-card__footer">
        <div className="movie-card__desc">
          <div className="movie-card__title">{movie.title}</div>
          <div className="movie-card__genre">{movie.genres.join(', ')}</div>
        </div>
        <div className="movie-card__date">{movie.release_date.slice(0, 4)}</div>
      </div>
    </div>
  );
}

export default MovieCard;
