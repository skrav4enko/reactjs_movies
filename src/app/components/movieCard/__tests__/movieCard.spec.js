/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import MovieCard from '..';

describe('Movie Card', () => {
  let testProps;

  beforeEach(() => {
    testProps ={
      navigateTo: jest.fn(),
      movie: {
        id: 338970,
        title: 'Tomb Raider',
        tagline: 'Her legend begins',
        vote_average: 6.2,
        vote_count: 817,
        release_date: '2018-03-08',
        poster_path: 'https://image.tmdb.org/t/p/w500/ePyN2nX9t8SOl70eRW47Q29zUFO.jpg',
        overview:
          'Lara Croft, the fiercely independent daughter of a missing adventurer.',
        budget: 94000000,
        revenue: 126025000,
        genres: ['Action', 'Adventure'],
        runtime: 118,
      }
    }
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<MovieCard {...testProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call navigateTo', () => {
    const component = shallow(<MovieCard {...testProps} />);
    const card = component.find('.movie-card').simulate('click');
    expect(testProps.navigateTo).toHaveBeenCalled();
  });
});
