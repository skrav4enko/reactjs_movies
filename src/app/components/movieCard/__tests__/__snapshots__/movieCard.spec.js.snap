// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Card should match snapshot 1`] = `
<index.tsx
  movie={
    Object {
      "budget": 94000000,
      "genres": Array [
        "Action",
        "Adventure",
      ],
      "id": 338970,
      "overview": "Lara Croft, the fiercely independent daughter of a missing adventurer.",
      "poster_path": "https://image.tmdb.org/t/p/w500/ePyN2nX9t8SOl70eRW47Q29zUFO.jpg",
      "release_date": "2018-03-08",
      "revenue": 126025000,
      "runtime": 118,
      "tagline": "Her legend begins",
      "title": "Tomb Raider",
      "vote_average": 6.2,
      "vote_count": 817,
    }
  }
  navigateTo={[MockFunction]}
/>
`;
